README.md

```diff
@@ -1,24 +1,256 @@
+```
+
+#### Enter a room
+```
+someCoroutineScope.launch{
+val result = controller.enter(yourPlayer,yourRoom)
+            when (result){
+                is RoomPresenceResult.Success -> //successful
+                is RoomPresenceResult.Failure -> //failure
+            }
+}
+```
+
+You must replace ```yourPlayer``` with your own instance of ```GamePlayer``` and ```yourGame``` with your own
+instance of ```GameRoom```
+
+#### Leave a room
+```
+someCoroutineScope.launch{
+val result = controller.leave(yourPlayer,yourRoom)
+            when (result){
+                is RoomPresenceResult.Success -> //successful
+                is RoomPresenceResult.Failure -> //failure
+            }
+}
+```
+You must replace ```yourPlayer``` with your own instance of ```GamePlayer``` and ```yourGame``` with your own
+instance of ```GameRoom```
+
+#### Send message to a room
+You can directly send a message to a room so that all room participants can register to and receive later
+```
+someCoroutineScope.launch {
+            val result = controller.sendMessageToRoom(player, room, GameMessage(messageContent = message))
+            when(result){
+                is MessageSentResult.Failed -> //message sent failed
+                is MessageSentResult.Success -> //message sent is success
+            }
+        }
+```
+
+#### Register to room messages
+
+```
+someCoroutineScope.launch {
+            controller.registerToRoomMessages(room,MessageType.TEXT).collect{ receivedMessage ->
+                // receivedMessage is received as ReceivedMessage
+            }
+        }
+```
+
+#### Register to player messages in the room
+
+If you want to receive messages from all players in room, you can simply use following
+```
+someCoroutineScope.launch {
+            controller.registerToPlayerMessagesInRoom(which, who, MessageType.TEXT).collect { receivedMessage ->
+                //received message from receivedMessage.from
+            }
+        }
+```
+
+#### Send message to another player
+```
+someCoroutineScope.launch {
+            val result = controller.sendMessageToPlayer(who, toWhom, GameMessage(messageContent = message))
+            when(result){
+                is MessageSentResult.Failed -> TODO()
+                is MessageSentResult.Success -> TODO()
+            }
+        }
+```
+
+#### Register to presence events in a room
+You can observe presence events in a room (currently supporting enter and leave) so you can react to it, for example
+notifying other players or updating players in your room.
+
+```
+controller.registerToPresenceEvents(which).collect { roomPresenceUpdate ->
+         //do something with roomPresenceUpdate
+      }
+```
+
+#### Query all players in a room
+You can query all players in a room like following. Notice that ```allPlayers()``` is a suspending function and needs
+ to be called from a coroutine scope.
+```
+someCoroutineScope.launch{
+   val allPlayers = allPlayers(room)
+
+}
+```
+
+
+
+
+
```

Comment:
Those blank lines can be removed :wink:

Reply:
https://github.com/ably-labs/GameRoomSDK/pull/8/commits/dd6af82937f959d3ec823a8824ebe449d91baec3

Reply:
https://github.com/ably-labs/GameRoomSDK/pull/8/commits/0f4b5c9a6d6f699f1936db0cc6638d096dac006e